addons/EasyNuke/EasyNuke.lua

```diff
@@ -0,0 +1,207 @@
+spell_tables["absorb"] = T{"Absorb-Acc","Absorb-TP","Absorb-Attri","Absorb-STR","Absorb-DEX","Absorb-VIT","Absorb-AGI","Absorb-INT","Absorb-MND","Absorb-CHR"}
+
+function find_index ()
+		if current_element == "fire" then elements_index = 1 end
+		if current_element == "wind" then elements_index = 2 end
+		if current_element == "thunder" then elements_index = 3 end
+		if current_element == "light" then elements_index = 4 end
+		if current_element == "ice" then elements_index = 5 end
+		if current_element == "water" then elements_index = 6 end
+		if current_element == "earth" then elements_index = 7 end
+		if current_element == "dark" then elements_index = 8 end
+end
+
+windower.register_event('addon command', function (command, arg)
+
+	if command == 'boom' then
+		if arg == nil then arg = 1 end
+		arg = tonumber(arg)
+		if arg > #spell_tables[current_element] then windower.send_command("@input /echo Invalid Spell. Try again.") return end
+		local current_spell_table = spell_tables[current_element]
+		windower.send_command("@input /ma "..current_spell_table[arg].." <"..target_mode..">")
+	end
+	
+	if command == "boomga" then
+		if arg == nil then arg = 1 end
+		arg = tonumber(arg)
+		if arg > #spell_tables[current_element]["ga"] then windower.send_command("@input /echo Invalid Spell. Try again.") return end
+		local current_spell_table = spell_tables[current_element]["ga"]
+		windower.send_command("@input /ma "..current_spell_table[arg].." <"..target_mode..">")
+	end
+	
+	if command == "boomra" then
+		if arg == nil then arg = 1 end
+		arg = tonumber(arg)
+		if arg > #spell_tables[current_element]["ra"] then windower.send_command("@input /echo Invalid Spell. Try again.") return end
+		local current_spell_table = spell_tables[current_element]["ra"]
+		windower.send_command("@input /ma "..current_spell_table[arg].." <"..target_mode..">")
+	end
+	
+	if command == "target" then
+		if arg then
+			arg = string.lower(arg)
+			target_mode = arg
+			windower.send_command("@input /echo Target mode is now: "..target_mode)
+		else targets_index = targets_index + 1
+		if targets_index > 3 then targets_index = 1 end
+		target_mode = targets[targets_index]
+		windower.send_command("@input /echo Target Mode is now: "..target_mode)
+		end
+	end
+	
+	if command == "element" then
+		arg = string.lower(arg)
+		if elements:contains(arg) then
+			current_element = arg
+			windower.send_command("@input /echo Element Mode is now: "..string.ucfirst(current_element))
+		elseif other_modes:contains(arg) then
+			current_element = arg
+			windower.send_command("@input /echo Element Mode is now: "..string.ucfirst(current_element))
+		else windower.send_command("@input /echo Invalid element")
+		end
+	end
+	
+	if command == "cycle" then
+		arg = string.lower(arg)
+		if arg == nil then
+			find_index()
+			elements_index = elements_index + 1
+			if elements_index > 8 then elements_index = 1 end
```

Comment:
If you took my suggestion from before these three lines could be combined into one:
```lua
            elements_index = indices[current_element] % 8 + 1
```
Doing this for the other places too will save you quite a few lines.